hokusai/services/deployment.py

```diff
@@ -74,12 +58,31 @@ def update(self, tag, constraint, git_remote, resolve_tag_sha1=True):
       print_green("Patching deployment %s..." % deployment['metadata']['name'])
       shout(self.kctl.command("patch deployment %s -p '%s'" % (deployment['metadata']['name'], json.dumps(patch))))
 
-    print_green("Waiting for rollout to complete...")
+    print_green("Waiting for deployment rollouts to complete...")
 
     rollout_commands = [self.kctl.command("rollout status deployment/%s" % deployment['metadata']['name']) for deployment in self.cache]
-    return_code = shout_concurrent(rollout_commands)
-    if return_code:
-      raise HokusaiError("Deployment failed!", return_code=return_code)
+    return_codes = shout_concurrent(rollout_commands, print_output=True)
+    if any(return_codes):
```

Comment:
This is where the deployments rollout magic happens - Hokusai is shelling out to `kubectl rollout status {deployment}` for each deployment in the project - if any of these commands fail, it will consider the entire deployment operation failed and move on to rollback by executing `kubectl rollout undo {deployment}` for each deployment. `kubectl rollout status` will return `1` if the deployment's `progressDeadlineSeconds` is exceeded.  For `apiVersion: extensions/v1beta1` type deployments, no `progressDeadlineSeconds` is set by default (hence the stalled deployments we have been seeing.  For `apiVersion: apps/v1` type deployments it is set to a default of `600` seconds.  So we should set this field everywhere or upgrade our deployment objects to  `apiVersion: apps/v1`. Details in this thread https://github.com/kubernetes/kubernetes/issues/66135